CRM/Core/Form/RecurringEntity.php

```diff
@@ -403,10 +404,11 @@ public static function postProcess($params, $type, $linkedEntities = []) {
           if (!empty(CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]['pre_delete_func']) &&
             !empty(CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]['helper_class'])
           ) {
-            $preDeleteResult = call_user_func_array(CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]['pre_delete_func'], [$params['entity_id']]);
-            if (!empty($preDeleteResult)) {
-              call_user_func([CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]['helper_class'], $preDeleteResult]);
-            }
+            // FIXME: This calls `CRM_Event_Form_ManageEvent_Repeat::checkRegistrationForEvents`
+            // which then sets the static variable `CRM_Core_BAO_RecurringEntity::$_entitiesToBeDeleted`
+            // which is then accessed below and used to delete events with no registrations.
+            // I can't think of a worse way to pass a variable back from a function.
+            call_user_func_array(CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]['pre_delete_func'], [$params['entity_id']]);
```

Comment:
If you find this confusing, you're not alone. L408 is completely nonsensical - that's just not how `call_user_func` works. All this does is crash. It's so broken that I honestly don't know what it was supposed to do, but here's my best guess:

1. All this muckety-muck about `CRM_Core_BAO_RecurringEntity::$_recurringEntityHelper[$params['entity_table']]['helper_class']` simply resolves to the string `CRM_Event_Form_ManageEvent_Repeat::checkRegistrationForEvents`
2. That function returns event ids that are safe to delete (that have no participants)
3. That function ALSO sets  the static variable `CRM_Core_BAO_RecurringEntity::$_entitiesToBeDeleted`
4. That variable is then used in the code below to (in an overcomplicated and difficult-to-read fashion) simply delete the events.
5. So I think this code is an earlier draft of the below code and was originally intended to do the same thing. I can't think of anything else it would be doing.
6. So I deleted it.

Perhaps the reason no one ever reported the problem is that it's a recoverable error. Refresh the page and everything appears ok.
But what bothers me a bit about this fix is that now that the crashey code is removed, it will longer prevents the subsequent code from running. Is that a good thing to enable "never worked" code? I think the intention is that as long as the repeating clones don't have anyone registered, they are safe to delete and rebuild. I *guess* that's ok...?